Update uses of cell_no and tile_no given new lake_cell_tile_xwalk

hcorson-dosch-usgs opened the pull request:
The new `lake_cell_tile_xwalk` [created in `lake-temperature-model-prep`](https://github.com/USGS-R/lake-temperature-model-prep/blob/main/_targets.R#L193-L205) contains two fields for both `cell_no` and `tile_no` -- a `spatial_X` and `data_X` field for each. The fields prefixed by `spatial_` indicate the cell and tile into which each lake falls spatially. The fields prefixed by `data_` indicate which cell and tile should be use for driver data for each lake. The `data_cell_no` will only differ from the `spatial_cell_no` for those lakes that fall within gcm cells that are missing data. The `data_tile_no` will only differ from the `spatial_tile_no` for those lakes that a) fall within gcm cells that are missing data, and b) are matched to a non-NaN cell in a different tile by our [matching code](https://github.com/USGS-R/lake-temperature-model-prep/blob/main/7_drivers_munge/src/GCM_driver_utils.R#L195-L242).

In this repo, we now use the `data_cell_no` instead of just `cell_no` to determine which meteo file to use for each lake model.

And for our [temporary solution](https://github.com/USGS-R/lake-temperature-process-models/issues/31) of zipping the output feathers by tile number, we now use `spatial_tile_no` instead of just `tile_no`, since we are grouping the output spatially and providing a map to guide users:
![image](https://user-images.githubusercontent.com/54007288/156795554-fc501ea7-5d5a-45de-9ebf-40a555301c5a.png)

hcorson-dosch-usgs committed:
Update uses of cell_no and tile_no to reflect changes to lake_cell_tile_xwalk with NA cell fix

lindsayplatt commented on 1_prep/src/build_model_config.R:
```diff
@@ -25,18 +28,19 @@ build_model_config <- function(meteo_feathers, lake_cell_tile_xwalk, gcm_names,
     cell_no = as.numeric(str_extract(tools::file_path_sans_ext(meteo_fl), paste0(cell_no_list, "$")))
```
Should this not be updated to make it clear if either the spatial or data cell is to be used? Then L42 in the join would join by `data_cell_no` and not need the distinction between left and right.

hcorson-dosch-usgs replied:
I was debating this. When we're working with the GCM data, we always use just `cell_no` (see [here](https://github.com/USGS-R/lake-temperature-model-prep/blob/main/7_drivers_munge/src/GCM_driver_utils.R#L505) and [here](https://github.com/USGS-R/lake-temperature-model-prep/blob/main/_targets.R#L219-L242)). And there is no `spatial_cell_no` vs `data_cell_no` when it comes to the GCM data itself. For that reason I wasn't sure if it would be confusing here to refer to the cell_no being pulled from the meteo file names as `data_cell_no`. But I can see that it could also be confusing to _not_ use `data_cell_no` here. And we are after all, pulling the `cell_no` that pertains to the meteo *data*. So I'll make that switch.

lindsayplatt commented on 1_prep.R:
```diff
@@ -19,8 +19,8 @@ p1 <- list(
-  tar_target(p1_site_ids, p1_lake_cell_tile_xwalk_df$site_id),
-  tar_target(p1_cell_nos, unique(p1_lake_cell_tile_xwalk_df$cell_no)),
+  tar_target(p1_site_ids, p1_lake_cell_tile_xwalk_df %>% pull(site_id)),
+  tar_target(p1_cell_nos, unique(p1_lake_cell_tile_xwalk_df %>% pull(data_cell_no))),
```
Note that the change from `$` to `pull()` was added because `$cell_no` returned `NULL` but didn't cause a failure because `cell_no` didn't exist. Using `pull()` throws an error when the column requested doesn't exist, which is the behavior we would rather see here.

lindsayplatt replied:
Writing this out so we have a record of it into the future. No need to do anything based on this comment.

hcorson-dosch-usgs committed:
refer to cell_no pulled from meteo file as data_cell_no

hcorson-dosch-usgs committed:
pull in upstream changes and fix merge conflicts

Merge branch 'main' of github.com:USGS-R/lake-temperature-process-models into update_cell_references

# Conflicts:
#	3_extract.R
#	3_extract/src/process_glm_output.R
